internal/controller/application_controller.go

```diff
@@ -163,49 +174,142 @@ func (r *ApplicationPullServiceAccountCreator) getImageRepositoryPullSecretNames
 	return pullSecretNames, nil
 }
 
-// createApplicationServiceAccount creates application service account with provided pull secrets
-func (r *ApplicationPullServiceAccountCreator) createApplicationServiceAccount(ctx context.Context, serviceAccountName string, application *appstudioredhatcomv1alpha1.Application, pullSecretNames []string) error {
-	log := ctrllog.FromContext(ctx).WithValues("serviceAccountName", serviceAccountName, "namespace", application.Namespace)
+// dockerConfigJson represents the structure of a .dockerconfigjson secret
+type dockerConfigJson struct {
+	Auths map[string]struct {
+		Auth string `json:"auth"`
+	} `json:"auths"`
+}
 
-	applicationServiceAccount := &corev1.ServiceAccount{}
-	err := r.Client.Get(ctx, types.NamespacedName{Name: serviceAccountName, Namespace: application.Namespace}, applicationServiceAccount)
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			log.Error(err, "failed to read application service account", l.Action, l.ActionView)
-			return err
+// createOrUpdateAggregatedPullSecret creates or updates a single kubernetes.io/dockerconfigjson secret
+// by combining data from individual pull secrets.
+func (r *AggregatedPullSecretCreator) createOrUpdateAggregatedPullSecret(ctx context.Context, aggregatedSecretName string, application *appstudioredhatcomv1alpha1.Application, individualSecretNames []string) (*corev1.Secret, error) {
+	log := ctrllog.FromContext(ctx)
+
+	combinedAuths := make(map[string]struct {
+		Auth string `json:"auth"`
+	})
+
+	for _, secretName := range individualSecretNames {
+		secret := &corev1.Secret{}
+		err := r.Client.Get(ctx, types.NamespacedName{Name: secretName, Namespace: application.Namespace}, secret)
```

Comment:
don't get each secret individually, do one query with corev1.SecretList{}

and then filter out those you need

Reply:
updated